app/controllers/users_controller.rb

```diff
@@ -21,10 +29,49 @@ def create
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+    return if @user
+    flash[:warning] = t :show
+    redirect_to root_path
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
+    if @user&.update_attributes user_params
+      flash[:success] = t :update
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    User.find_by(params[:id]).destroy
+    flash[:success] = t :delete
+    redirect_to users_url
+  end
+
   private
 
   def user_params
     params.require(:user).permit :name, :email, :password,
       :password_confirmation
   end
+
+  def logged_in_user
+    unless logged_in?
+      flash[:danger] = t :log_in
+      redirect_to login_url
+    end
+  end 
+
+  def correct_user
+    @user = User.find_by id: params[:id]
+    redirect_to root_url unless current_user? @user
+  end
+
+  def admin_user
```

Comment:
Bad naming. Please rename it convenient with it's functionality, like as verify_admin/verify_admin_user